zerver/lib/markdown/fenced_code.py

```diff
@@ -392,6 +393,13 @@ def format_code(self, lang: str, text: str) -> str:
         else:
             code = CODE_WRAP.format(langclass, self._escape(text))
 
+        # In order to display a "view-in-playground" option in the frontend,
+        # we need to know the language used in the codeblock. We tweak the HTML
+        # CodeHilite generates to add this language as a data-attribute.
+        if lang:
+            parsed_code = BeautifulSoup(code, features='lxml')
+            parsed_code.find('div')['data-codehilite-language'] = lang
```

Comment:
We should probably add some comments here that make it a bit more clear where that div is coming from.  It's not coming from CodeHilite, is it?  Here is the relevant code from CodeHilite: https://github.com/Python-Markdown/markdown/blob/master/markdown/extensions/codehilite.py#L158

Reply:
I think `self.use_pygments` is true and it doesn't use that code path.  See the `css_class` option docs for some detail on how this is configured.